lib/omni-sharp-server/client-manager.ts

```diff
@@ -12,11 +18,44 @@ class ClientManager {
     private _projectPaths: string[] = [];
     private _activated = false;
     private _temporaryClients: { [path: string]: RefCountDisposable } = {};
+    private _omnisharpAtom: typeof OmniSharpAtom;
+    private _activeClients: Client[] = [];
+
+    private _aggregateClient = new CompositeClient();
+    public get aggregateClient() { return this._aggregateClient; }
+
+    public _clientsSubject = new ReplaySubject<Client[]>(1);
+    private _clientStateObservable = this._clientsSubject
+        .flatMap(x => Observable.combineLatest(x.map(z => z.state), function(...values) { return values; }));
+
+    private _isOff = true;
+    public get isOff() { return this._isOff; }
+    public get isOn() { return !this.isOff; }
+
+    public get state() { return this._clientStateObservable; }
+
+    private _activeClient = new ReplaySubject<Client>(1);
+    public get activeClient(): Observable<Client> { return this._activeClient; }
+
+    public get numberOfClients() {
+        return this._clientPaths.length;
+    }
+
+    constructor() {
+        this._clientsSubject.onNext(this._activeClients);
```

Comment:
Can you add a comment on why its necessary to for to subject to be signalled once in the constructor of the ClientManager (which we use as a singleton)